timelock/program/src/state/mod.rs

```diff
@@ -0,0 +1,104 @@
+use crate::{
+    error::LendingError,
+    math::{Decimal, Rate, SCALE},
+};
+use arrayref::{array_mut_ref, array_ref, array_refs, mut_array_refs};
+use solana_program::{
+    clock::{Slot, DEFAULT_TICKS_PER_SECOND, DEFAULT_TICKS_PER_SLOT, SECONDS_PER_DAY},
+    entrypoint::ProgramResult,
+    program_error::ProgramError,
+    program_option::COption,
+    program_pack::{IsInitialized, Pack, Sealed},
+    pubkey::Pubkey,
+    sysvar::clock::Clock,
+};
+
+const TRANSACTION_SLOTS: u8 = 10;
+const TIMELOCK_VERSION: u8 = 1;
+pub const INSTRUCTION_LIMIT: u64 = 2_000_000;
+
+pub enum ConsensusAlgorithm {
```

Comment:
Add Enum for different transaction types:

Token Swap
Swap Buffer Instruction 
Eventually - Custom Instructions (unsupported)

State that goes with each type is specific for each type of thing...easier to build for now.